[W17-4] DeadlineManager

btzy opened the pull request:
@sijie123 @sidhant007 @ranaldmiao @kc97ble

parvathy-sudhir-pillai commented on README.adoc:
```diff
@@ -15,7 +15,7 @@ ifndef::env-github[]
+* This is a desktop deadline manager application. It has a GUI but most of the user interactions happen using a CLI (Command Line Interface).
 * It is a Java sample application intended for students learning Software Engineering while using Java as the main programming language.
 * It is *written in OOP fashion*. It provides a *reasonably well-written* code example that is *significantly bigger* (around 6 KLoC)than what students usually write in beginner-level SE modules.
 * What's different from https://github.com/se-edu/addressbook-level3[level 3]:
```
Can possibly remove the links to AB?

parvathy-sudhir-pillai commented on docs/UserGuide.adoc:
```diff
@@ -24,7 +24,7 @@ AddressBook Level 4 (AB4) is for those who *prefer to use a desktop app for mana
 .  Ensure you have Java version `9` or later installed in your Computer.
 .  Download the latest `addressbook.jar` link:{repoURL}/releases[here].
-.  Copy the file to the folder you want to use as the home folder for your Address Book.
+.  Copy the file to the folder you want to use as the home folder for your deadline manager.
```
Most of the adocs still are titled as AddressBook - Level 4. Change these to reflect your app's.
Changes at this point, do not reflect how you intend to extend AB4

parvathy-sudhir-pillai reviewed (commented):
Good effort, team! 

Suggestions to improve:

* Preview how your adocs get rendered using the netlify preview.
* Most of the adocs still are titled as AddressBook - Level 4. Change these to reflect your app's
* Remove unnecessary/dead links. The View on GitHub link does not work.
* Remove the header links which are actually from AB4.
* Target user profile, value proposition are still from AB4.
* Change use cases to those of your apps'.

joanneong commented on docs/UserGuide.adoc:
```diff
@@ -46,57 +49,88 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 
 [TIP]
-A person can have any number of tags (including 0)
+You can ask two tasks with the same attributes!
```
"...ask two tasks with..." 🤔

joanneong commented on docs/UserGuide.adoc:
```diff
@@ -46,57 +49,88 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
+* `add n/Assignment 2 d/1/1/2018 p/1` +
+Adds a task with name `Assignment 2` with a deadline on 1st January 2018 with priority 1 (highest priority).
+* `add n/v1.1 milestone d/9/10/2018 t/CS2103T t/Project p/2` +
+Adds a task with name `v1.1 milestone` with a deadline on 9th October 2018 with priority 2 (second highest priority). It is additionally tagged with 2 tags: `CS2103T1 and `Project`.
```
Slight formatting error here (try and see if you can find it)

joanneong commented on docs/UserGuide.adoc:
```diff
@@ -109,50 +143,230 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
+Format: `resolve INDEX`
+
+****
+* Hides the task at the specified INDEX. The index refers to the index number shown in the displayed task list. The index must be a positive integer 1, 2, 3, …​
```
"Hide" implies that you can retrieve it again 🤔

joanneong reviewed (commented):
Keep up the good progress! :)

joanneong commented on README.adoc:
```diff
@@ -15,24 +16,23 @@ ifndef::env-github[]
+* Our main value proposition:
+** Fully featured command line interface, allowing users to do what they need faster without needing to use the mouse.
+** More features than regular deadline managers, such as the ability to snooze impending deadlines and procrastinate.
+** Fully tested, including automated GUI testing.
```
I am not sure if I would consider full testing as a kind of value proposition 🤔

joanneong commented on docs/UserGuide.adoc:
```diff
@@ -109,50 +145,281 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
+Returns a subset of the current list of tasks that have priority = 1 (highest priority).
+
+* `filter p<3`
+Returns a subset of the current list of tasks that have priority < 3. (Highest priority or second highest priority.)
```
Wouldn't this also return tasks with priority = 3?

joanneong commented on docs/UserGuide.adoc:
```diff
@@ -109,50 +145,281 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
+
+****
+* In case a task does not belong to any of the tags mentioned it will be placed at the bottom of the sorted list.
+* In case a task contains more than one tag specified in the comparator, then it will be attempted to be sorted according to those tags first which place it former in the sorted list.
```
This sentence doesn't sound right.

joanneong commented on docs/UserGuide.adoc:
```diff
@@ -109,50 +145,281 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
+* `export cs2103.txt` +
+Exports tasks to the file `cs2103.txt` in the same folder as Deadline Manager.
+* `export C:\Documents\cs2103.txt`
+Export tasks to the file `cs2103.txt` that is defined by the absoluted path provided.
```
Slight formatting issue here (render and check)

joanneong commented on docs/UserGuide.adoc:
```diff
@@ -109,50 +145,281 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
+=== Deleting a task : `delete`
+
+Deletes the specified task from the deadline manager. +
 Format: `delete INDEX`
```
What is the difference between this command and the resolve command?

joanneong commented on docs/UserGuide.adoc:
```diff
@@ -109,50 +145,281 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
-Deletes the 2nd person in the address book.
-* `find Betsy` +
+Deletes the 2nd task in the deadline manager.
+* `find `OP1 Assignment` +
```
There seems to be a slight typo here.

btzy replied:
There does not seem to be any typo here.

joanneong replied:
If you look at it via Netlify you will see what I mean :P

btzy replied:
Okay, resolved.

joanneong commented on docs/UserGuide.adoc:
```diff
@@ -226,35 +495,87 @@ Format: `exit`
+* *Add Recurring*: `addr n/NAME d/DEADLINE f/FREQUENCY [P/PRIORITY] [t/TAG]...`
+e.g. `add n/CS2103 Tutorial d/1/10/2018 f/7 P/4 t/easy`
+
+* *Attachment (Add)* : `attachment INDEX add p\FILEPATH`
```
Why is it `\p` here...

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -11,15 +11,41 @@ ifdef::env-github[]
+By: `Team W17-4`      Since: `Sep 2018`      Licence: `MIT`
+
+The Deadline Manager is an app to help users keep track of upcoming deadlines. It is targeted for students in NUS School of Computing, and has features dedicated to improve their workflow.
+This *Developer Guide* help you get started with contributing to this open source project.
```
This sentence is grammatically flawed.

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -157,7 +203,24 @@ image::SDforDeletePersonEventHandling.png[width="800"]
 [NOTE]
```
The diagram has a very slight formatting issue with one of the closing brackets.

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -187,16 +252,23 @@ image::LogicClassDiagram.png[width="800"]
 .Interactions Inside the Logic Component for the `delete 1` Command
 image::DeletePersonSdForLogic.png[width="800"]
 
+.  `Logic` uses the `TaskCollectionParser` class to parse the delete command.
```
Can you update the diagram above to reflect this?

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -225,72 +299,80 @@ image::StorageClassDiagram.png[width="800"]
+
+=== Add feature
+
+Working in progress.
```
Please remember to update this section!

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -302,33 +384,424 @@ image::UndoRedoActivityDiagram.png[width="650"]
+=== Import and Export feature
+//==== Current implementation
+
+The deadline supports import and export features. This feature uses 2 components: `Storage` and `Model`.
```
"The deadline supports..." 🤔

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -302,33 +384,424 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+Given below is a sequence diagram illustrating the interactions between `Model` and `Storage` when importing.
+
+image::ImportSequenceDiagram.png[width="800"]
```
When self-calling a method in `Model`, it seems like the rectangle has merged into the main activation bar

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -302,33 +384,424 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+* `IgnoreResolver`: Discards the incoming task in favour of the existing task.
+* `OverwriteResolver`: Updates the existing task to reflect the values of the incoming task.
+* `DuplicateResolver`: Keep both copies.
```
I don't actually remember seeing this in the User Guide 🤔

joanneong replied:
Also, any noteworthy design considerations to share for this feature?

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -302,33 +384,424 @@ image::UndoRedoActivityDiagram.png[width="650"]
+Step 4. The `SortCommand` calls the `updateSortedTaskList` API method of `ModelManager`
+
+Step 5. The method `updateSortedTaskList` sorts the `versionTaskCollection` using JavaFx sort method
+
```
Here is a good opportunity to show that you know sequence diagrams

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -302,33 +384,424 @@ image::UndoRedoActivityDiagram.png[width="650"]
+The attachment feature aims to help users organise and keep track of important files needed for the respective tasks. It allows users to associate files in their computer with a particular task. When the user decides to perform the task, this feature provides functionality to identify previously associated files and export it to an appropriate location.
+
+==== Current implementation
+The attachment feature is mainly implemented by `AttachmentCommand`. As `AttachmentCommand` is an all-purpose command that allows users to perform numerous actions related to attachments, a nested abstract class `AttachmentAction` is defined within `AttachmentCommand`. There are 4 classes that extend `AttachmentAction` in order to implement the various actions of the attachment feature. The comma ...
```
Can you break the second statement into two?

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -302,33 +384,424 @@ image::UndoRedoActivityDiagram.png[width="650"]
+Step 3. The `Model` is updated with the `Task` object returned by the `perform` method in Step 2.
+
+Step 4. The `resultMessage` method of `AttachmentAction` is invoked. The returned value is used to construct a `CommandResult` object and returned to the `LogicManager`.
+
```
This section is a good place to introduce sequence diagram(s)

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -302,33 +384,424 @@ image::UndoRedoActivityDiagram.png[width="650"]
+* **Alternative 2:** Use the path to the file as an identifier.
+** Pros: Allows for more than one attachments with the same file name to be associated with the same task. For example: `folder1/Template.docx` and `folder2/Template.docx` can be added to the same task as attachments.
+** Cons: Cumbersome for the user to type the full path to identify the file.
+
```
Good design considerations 👍

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -833,33 +1314,33 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 |Priority |As a ... |I want to ... |So that I can...
```
Update your target user profile to match your product!

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -833,33 +1314,33 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
+|`*` |user with many tasks in the deadline manager |sort tasks by name |locate a task easily
 |=======================================================================
 
```
Are you sure these are all the user stories that you have?

joanneong replied:
Use cases as well below

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -883,7 +1364,7 @@ _{More to be added}_
+.  Should be able to hold up to 1000 tasks without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 
 _{More to be added}_
```
Is the glossary updated for your application?

joanneong reviewed (commented):
Good job in general. I can see the effort made to update the documentation. However, it seems like someone in the team has not updated the developer guide for their feature yet - let me know after you do so and I will take a look.

Also, what happened to your Travis? Why is it not triggering here? (oh cos you have conflicts)

Sorry for the delayed review!

btzy commented:
@joanneong Merge conflict with nus-cs2103-AY1819S1/addressbook-level4:master has been resolved.  Travis should work now.

btzy committed:
Update user guide and developer guide for filter command

btzy committed:
DeveloperGuide: Add description for filter error processing

btzy committed:
Merge pull request #264 from btzy/filter-ugdg

Update UG and DG for filter

sijie123 committed:
ImportExport: Support undo for import (export has no undo feature).

sijie123 committed:
Merge pull request #265 from sijie123/undoimport

ImportExport: Support undo for import

sijie123 committed:
Add more unit tests

sijie123 committed:
Fix import order and conform to style guide

sijie123 committed:
Add unit test to StorageManager, add checks to ensure non null in StorageManager

sijie123 committed:
Merge pull request #266 from sijie123/undoimport

Add more unit tests

btzy committed:
Add more diagrams to developer guide

btzy committed:
Merge pull request #267 from btzy/filter-ugdg

Add more diagrams to developer guide

btzy committed:
Adhere to CS2103 Java coding style

btzy committed:
Merge pull request #268 from btzy/fix-coding-style

Adhere to CS2103 Java coding style

btzy committed:
Filter: Add better inline help

This inline help is displayed to the user when `filter` is typed without any filter expression.

btzy committed:
Fix style issues

btzy committed:
Remove redundant import

btzy committed:
Merge pull request #269 from btzy/filter-command-help

Filter: Add better inline help

btzy committed:
UserGuide: State that filter will be reset on any modifying command

btzy committed:
DeveloperGuide: State that filter will reset on any modifying command

btzy committed:
Merge pull request #271 from btzy/filter-ugdg

State that filter will reset on any modifying command

btzy committed:
Deadline: Allow dates without year, disallow current-century adjustment

btzy committed:
UserGuide: Add explanation on deadline format

btzy committed:
More tests for deadline

btzy committed:
Remove unused import

btzy committed:
UserGuide: Explicitly state constraints on deadline year value

btzy committed:
Merge pull request #273 from btzy/deadline

Improve deadline parsing

btzy committed:
UserGuide: Fix incorrect .jar file name

btzy committed:
Merge pull request #274 from btzy/filter-ugdg

Fix incorrect .jar file name in user guide

btzy committed:
UserGuide: Recommend quoting for argument tokenization

btzy committed:
ArgumentTokenizer: Convert some methods to private

btzy committed:
Merge pull request #276 from btzy/argument-tokenizer-javadoc

Argument tokenizer javadoc

btzy committed:
Add some logging for filter

btzy committed:
Improve SLAP for filter command and related classes

btzy committed:
Adhere to more code quality guidelines

btzy committed:
UserGuide: Highlight that simplified filter syntax is usually sufficient

btzy committed:
UserGuide: Increase size of diagram for filter

btzy committed:
Merge pull request #277 from btzy/filter-code-quality

[WIP] Improve code quality for filter

btzy committed:
DeveloperGuide: Highlight simplification of filter diagram

btzy committed:
UserGuide & DeveloperGuide: Update filter specification

ranaldmiao committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-W17-4/main

btzy committed:
UG & DG: Improve readability for filter

ranaldmiao committed:
Update UG

Minor edit for attachments.
Removed 'resolve' feature completely.
Updated examples for add command to working commands.

btzy committed:
Merge pull request #275 from btzy/filter-ugdg

UG & DG: Updates to conform to the current implementation

ranaldmiao committed:
Update UG

Minor edit for Quick Start.

ranaldmiao committed:
Update UG - remove 'snooze' command.

btzy committed:
Refractor field identifiers from FilterCommandParser to CliSyntax

btzy committed:
Removed usused export constant from CliSyntax

ranaldmiao committed:
Update UG - attachments feature

Add specific command format for each actions of AttachmentCommand.

btzy committed:
ParseException: Improve eager error checking

btzy committed:
Reorder imports for FilterCommandParser to suit style

ranaldmiao committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-W17-4/main

# Conflicts:
#	docs/UserGuide.adoc

btzy committed:
Merge pull request #280 from btzy/parseexception-preconditions

ParseException: Improve eager error checking

btzy committed:
UserGuide: Add internal links for filter

btzy committed:
Merge pull request #281 from btzy/filter-ugdg

UserGuide: Add internal links for filter

btzy committed:
UserGuide: Mention that DeadlineManager is not timezone-aware

ranaldmiao committed:
Merge pull request #279 from ranaldmiao/master

Update User Guide

btzy committed:
Merge pull request #282 from btzy/filter-ugdg

UserGuide: Mention that DeadlineManager is not timezone-aware

btzy committed:
Merge pull request #278 from btzy/extract-shared-field-identifiers

Refractor field identifiers from FilterCommandParser to CliSyntax

ranaldmiao committed:
Update Developer Guide

Refactor AttachmentCommand:
AttachmentAction is now an interface instead of an abstract class

Developer Guide:
Added snippets of code for Attachment feature
Added use case in Appendix D
Cleaned up a few other appendices

ranaldmiao committed:
Merge branch 'main'

ranaldmiao committed:
Update Ranald's PPP

ranaldmiao committed:
Merge pull request #283 from ranaldmiao/master

Update Developer Guide

ranaldmiao committed:
Merge pull request #284 from ranaldmiao/master

Update Ranald's PPP

ranaldmiao committed:
Update Ranald's PPP

Trying to cut content and the number of pages

ranaldmiao committed:
Merge pull request #285 from ranaldmiao/master

Update Ranald's PPP

sijie123 committed:
Remove duplicate check. User can add any task to the deadline manager, even if the same task currently exists.

sijie123 committed:
Update test to match coding style

sijie123 committed:
Update Si Jie's Import and Export features

Reword import and export features

sijie123 committed:
Merge branch 'update_userguide' of https://github.com/CS2103-AY1819S1-W17-4/main into ug

sijie123 committed:
Update user guide

sijie123 committed:
Merge pull request #286 from sijie123/master

Remove duplicate check. User can add any task to the deadline manager…

btzy committed:
DeveloperGuide: Clarify save/restore of string tokenizer

sijie123 committed:
Merge pull request #287 from CS2103-AY1819S1-W17-4/update_userguide

Update Si Jie's Import and Export features

btzy committed:
Add introductory paragraph to PPP for Bernard

btzy committed:
Add link to project code dashboard for Bernard

sidhant007 committed:
Fix incomplete tag issue

sidhant007 committed:
Filter empty strings during tag comparison

sidhant007 committed:
Fix style

sidhant007 committed:
Merge pull request #291 from sidhant007/fix-bug

Fix incomplete tag command for sort

sidhant007 committed:
Update tests

sidhant007 committed:
Merge pull request #292 from sidhant007/allow-spaces

Allow spaces for tags in sort command

btzy committed:
Extract exerpts from UG and DG for Bernard's PPP

btzy committed:
Merge pull request #288 from btzy/filter-ugdg

Project portfolio page for Bernard

ranaldmiao committed:
Update Ranald's PPP

Cutting content to fit 10 pages

ranaldmiao committed:
Update Attachment's tip

To be in sync with Import/Export regarding quotation marks for filename and filepath

ranaldmiao committed:
Merge pull request #294 from ranaldmiao/master

Update Ranald's PPP and User Guide for Attachments

ranaldmiao committed:
Update Ranald's PPP

Reformat more content to fit within page limit.

ranaldmiao committed:
Merge pull request #295 from ranaldmiao/master

Update Ranald's PPP

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -230,72 +299,92 @@ image::StorageClassDiagram.png[width="800"]
+or shift the deadline to the next occurrence
+(if the task is recurred). Thanks to this command,
+the user no longer needs to decide if he/she needs
+to use `delete` command or `snooze` command.
```
As written at the start of this section:
```
This section describes some noteworthy details on how major features are implemented.
```
So actually it would be fine not to include a section on `Add feature` and `Complete feature` since no implementation details are found in these sections either

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -307,33 +396,653 @@ image::UndoRedoActivityDiagram.png[width="650"]
+Step 3. The `SortCommandParser` parses the string into a valid comparator and calls the `SortCommand` with this comparator provided.
+
+image::SortOperationSequenceDiagram.png[width="800"]
+Figure 10 illustrates the sequence diagram for Sort Command
```
There should not still be an activation bar right after the `SortCommand` constructor is called and returned. Also, the vertical bar in the constructor is typically hidden behind the horizontal bar

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -307,33 +396,653 @@ image::UndoRedoActivityDiagram.png[width="650"]
+
+A general string tokenizer (this is the `StringTokenizer` class in the `parser` subdirectory) is shared by parts 1 and 2 to split the filter expression into tokens.
+
+image::FilterOperationSequenceDiagram.png[width="800"]
```
Nice diagram 👍

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -307,33 +396,653 @@ image::UndoRedoActivityDiagram.png[width="650"]
+** Pros: Simplest method signatures.
+** Cons: Easy to overlook possible failure conditions when modifying the code.
+
+// end::filter-design-considerations-extract[]
```
This section is **incredibly** detailed. Amazing effort put in!

joanneong commented on docs/DeveloperGuide.adoc:
```diff
@@ -307,33 +396,653 @@ image::UndoRedoActivityDiagram.png[width="650"]
+Given below is another sequence of steps, also illustrated by the sequence diagram. They describe the interaction between `AttachmentCommand` and classes that implement `AttachmentAction` after the `AttachmentCommand#execute` method is invoked by the `LogicManager`.
+
+.Sequence Diagram for performing the `execute` method of an `AttachmentCommand`
+image::AttachmentExecuteSequenceDiagram.png[width="800"]
```
Choice of whether to insert return arrows seems arbitrary for this diagram...

ranaldmiao replied:
Will it be okay if I just add the description of the return arrow for getFilteredTaskList?
And a elaboration that return arrows that are omitted are for 'void' methods where there are no return value.

![image](https://user-images.githubusercontent.com/2166767/48310050-98372600-e5c1-11e8-9cb5-59551e39b7e6.png)

joanneong replied:
Sure

joanneong reviewed (commented):
Just a final quick developer guide review. Excellent effort made in general, with some very nice diagrams :)

Some minor comments given. Also, it seems that the figure numbering messed up midway through the guide...

ranaldmiao committed:
Update Dev Guide

Add instructions for manual testing for Attachment Feature.

ranaldmiao committed:
Update Dev Guide

Update AttachmentCommand sequence diagram

Added return arrows

ranaldmiao committed:
Merge pull request #296 from ranaldmiao/master

Update Dev Guide

btzy committed:
DeveloperGuide: Fix minor errors

ranaldmiao committed:
Update Dev Guide

Update AttachmentCommand sequence diagram pptx

ranaldmiao committed:
Merge pull request #299 from ranaldmiao/master

Update Dev Guide

ranaldmiao committed:
Refactor AttachmentCommand

Fix #248 as well

ranaldmiao committed:
Fix style

btzy committed:
Add figure labels for diagrams

ranaldmiao committed:
Fix style

ranaldmiao committed:
Delete DuplicateTaskException

Deadline Manager allows duplicated tasks so this exception is useless.

ranaldmiao committed:
Merge pull request #300 from ranaldmiao/master

Refactor AttachmentCommand

ranaldmiao committed:
Add limits on frequency

Frequency can now only be from 1 to 999 inclusive.

ranaldmiao committed:
Update User Guide to specify limit on frequency.

ranaldmiao committed:
Fix valid frequency check to allow frequency 0.

ranaldmiao committed:
Fix style

btzy committed:
Merge pull request #297 from btzy/filter-ugdg

DeveloperGuide: Fix minor errors and add figure labels

ranaldmiao committed:
Add FrequencyTest

ranaldmiao committed:
Fix style

ranaldmiao committed:
Add more tests for Frequency

sijie123 committed:
Update Developer Guide for Sijie's Import/Export

ranaldmiao committed:
Merge pull request #301 from ranaldmiao/master

Delete DuplicateTaskException

btzy committed:
Add sample tasks

ranaldmiao committed:
Add more tests for Frequency

btzy committed:
Merge pull request #304 from btzy/add-sample-tasks

Add sample tasks

sijie123 committed:
Conform with Travis CI adoc style checks

ranaldmiao committed:
Remove redundant < 0 check for Frequency.

sidhant007 committed:
Updated UG+DG+PPP

sidhant007 committed:
Minor bug fixes

sijie123 committed:
Merge pull request #303 from sijie123/ug

Update Developer Guide for Sijie's Import/Export

sidhant007 committed:
Merge pull request #305 from sidhant007/bug-fix2

Minor bug fixes

btzy committed:
Add instructions for manual testing for filter

btzy committed:
Merge pull request #306 from btzy/filter-ugdg

DeveloperGuide: Add instructions for manual testing for filter

sijie123 committed:
Update DG for Import/Export to make it clearer.

btzy committed:
Shorten PPP for Bernard

btzy committed:
Merge pull request #308 from btzy/filter-ugdg

Shorten PPP for Bernard

sidhant007 committed:
Update PPP

sidhant007 committed:
Minor fix

sidhant007 committed:
Remove magic constants

sidhant007 committed:
Merge branch 'master' into ppp-ug-dg

sidhant007 committed:
Merge pull request #309 from sidhant007/remove-magic-constants

Remove magic constants from sort

sidhant007 committed:
Merge pull request #310 from sidhant007/ppp-ug-dg

Update UG + DG + PPP (Sidhant)

ranaldmiao committed:
Merge branch 'main' into freq-fix

sidhant007 committed:
Add newline to enable rendering of images

sidhant007 committed:
Merge pull request #311 from sidhant007/bug-fix3

Minor UG Fix

sijie123 committed:
Merge pull request #307 from sijie123/ug

Update DG for Import/Export to make it clearer.

sijie123 committed:
Add checks to original master

btzy committed:
Make 'fallthrough' capitalized for match style

ranaldmiao committed:
Refactor attachment

btzy committed:
Merge pull request #314 from btzy/fallthrough

Make 'fallthrough' capitalized for match style

sijie123 committed:
Change indent to conform to OSS Coding Style

sijie123 committed:
Update tests to conform to OSS Coding Standards

ranaldmiao committed:
Fix style check for wrapped lines

sijie123 committed:
Conform to Coding Style Standards for Lambda expressions

sijie123 committed:
Merge pull request #313 from sijie123/master

[WIP] Add checks to import export storage

kc97ble committed:
Update UG for ADD and COMPLETE command

ranaldmiao committed:
Fix style check for wrapped lines

ranaldmiao committed:
Update checkstyle

Change line continuation to +8 (thanks sijie123)

ranaldmiao committed:
Refactor many more files to update indentation

ranaldmiao committed:
Update indentation for Command Tests

ranaldmiao committed:
Update indentation for Parser Tests

ranaldmiao committed:
Update indentation for Unit tests

ranaldmiao committed:
Update indentation for system tests

sidhant007 committed:
Add java code

sidhant007 committed:
Merge pull request #316 from sidhant007/update-ppp

Update PPP (Add java code)

sidhant007 committed:
Make fallthrough capitalised

sidhant007 committed:
Change url for PPP commits page

sidhant007 committed:
Merge pull request #317 from sidhant007/follow-style

Make fallthrough capitalized to follow style guide

kc97ble committed:
Improve DG for COMPLETE command

sidhant007 committed:
Minor fix

sidhant007 committed:
Merge pull request #318 from sidhant007/minor-fix4

Minor fix to Sidhant PPP

kc97ble committed:
Update DG for ADD command

kc97ble committed:
Add figures to DG for ADD and COMPLETE commands

kc97ble committed:
Add figures in UG for COMPLETE command

kc97ble committed:
Merge pull request #319 from kc97ble/kien-ugdg-1

Update UG, DG for ADD and COMPLETE commands

kc97ble committed:
Update Kien's PPP

ranaldmiao committed:
Merge branch 'main' into freq-fix

# Conflicts:
#	docs/UserGuide.adoc

ranaldmiao committed:
Merge branch 'kien-ppp-1' into freq-fix

ranaldmiao committed:
Fix help text for invalid frequency.

btzy committed:
Revert changes to FilterCommandParser

btzy committed:
Fix indentation

btzy committed:
Refractor some code in DeadlineDateFormat

btzy committed:
Update PPP for Bernard

btzy committed:
Update PPP for Bernard

btzy committed:
Fix error in PPP

btzy committed:
UserGuide: Add some screenshots for filter

sijie123 committed:
Update sijie123 ppp

sijie123 committed:
Merge pull request #323 from sijie123/ppp

Update sijie123 ppp

btzy committed:
UserGuide: Add composite filter screenshot

btzy committed:
Remove trailing whitespace

btzy committed:
Add simplified filter syntax to PPP for Bernard

btzy committed:
UserGuide tag adjustments

btzy committed:
Fix grammar error

sidhant007 committed:
Fix numbering of figures

sidhant007 committed:
Merge pull request #324 from sidhant007/fix-figure-numbering

Fix numbering of figures [Sort]

btzy committed:
Fork UG and DG for Bernard's PPP, shorten to 10 pages

sidhant007 committed:
Fix grammar

btzy committed:
Merge pull request #321 from btzy/filter-ugdg

Add screenshots for filter and update PPP for Bernard

kc97ble committed:
Merge pull request #320 from kc97ble/kien-ppp-1

Update Kien's PPP

sidhant007 committed:
mend

btzy committed:
Fix frequency error message to say "integer" not "number"

sidhant007 committed:
Make priority format and frequency format more visible to user

sijie123 committed:
Update sijie ppp to cut space

sijie123 committed:
Merge pull request #327 from sijie123/ppp

Update sijie ppp to cut space

btzy committed:
Merge pull request #326 from btzy/fix-frequency-message

Fix frequency error message to say "integer" not "number"

sidhant007 committed:
Update definition of frequency

sidhant007 committed:
Merge pull request #325 from sidhant007/fix-grammar

Fix grammar

btzy committed:
UserGuide: Fix grammar issues and improve clarity

sijie123 committed:
Cut even more words

sijie123 committed:
Merge branch 'master' into updateIndentation

sijie123 committed:
Merge pull request #329 from sijie123/ppp

Cut even more words for Si Jie's PPP

btzy committed:
DeveloperGuide: Fix grammar issues and improve clarity

btzy committed:
Merge pull request #328 from btzy/filter-ugdg

Fix grammar issues and improve clarity

sijie123 committed:
Add ImportExport System Test

kc97ble committed:
Update Instructions for Manual Testing for COMPLETE command

ranaldmiao committed:
Update Ranald PPP

ranaldmiao committed:
Update Ranald PPP

ranaldmiao committed:
Merge pull request #333 from ranaldmiao/ppp

Update Ranald PPP

ranaldmiao committed:
Merge branch 'master' into freq-fix

btzy committed:
Update some diagrams for filter

ranaldmiao committed:
Update Ranald PPP

Fix typo

ranaldmiao committed:
Merge pull request #337 from ranaldmiao/ppp

Update Ranald PPP

sijie123 committed:
Merge pull request #331 from sijie123/systemTestImportExport

Add ImportExport System Test

sijie123 committed:
Merge pull request #302 from ranaldmiao/freq-fix

Add limit for the maximum frequency of a task

btzy committed:
Update diagrams and some explanation in DG

sijie123 committed:
Update sijie PPP

sijie123 committed:
Merge pull request #338 from sijie123/systemTestImportExport

Update sijie PPP

ranaldmiao committed:
Update Ranald PPP

Add back design considerations
Add back more content

kc97ble committed:
Merge pull request #334 from kc97ble/kien-ppp-2

Update Instructions for Manual Testing for COMPLETE command

sijie123 committed:
Fix small typo in developer guide

ranaldmiao committed:
Merge pull request #339 from ranaldmiao/ppp

Update Ranald PPP

sijie123 committed:
Fix "adding a task" of the manual testing section in the DG

btzy committed:
Minor changes to filter section

btzy committed:
Adjust items to make PPP 10 pages long for Bernard

btzy committed:
Revert checkstyle

ranaldmiao committed:
Merge branch 'master' into updateIndentation

sijie123 committed:
Cut lines for sijie PPP

sijie123 committed:
Merge branch 'master' into systemTestImportExport

sijie123 committed:
Merge pull request #340 from sijie123/systemTestImportExport

Update docs

btzy committed:
Merge pull request #335 from btzy/fix-dg-diagrams

Minor fixes to filter UML diagrams in DG

btzy committed:
Remove alternate stylesheet

ranaldmiao committed:
Merge branch 'main' into updateIndentation

btzy committed:
Merge pull request #341 from btzy/fix-dg-diagrams

Remove alternate stylesheet

sijie123 committed:
Merge branch 'systemTestImportExport' of https://github.com/sijie123/CS2103-DeadlineManager into systemTestImportExport

# Conflicts:
#	docs/DeveloperGuide.adoc

ranaldmiao committed:
Fix style for new styleguide

ranaldmiao committed:
Merge branch 'main' into updateIndentation

sijie123 committed:
Merge pull request #342 from sijie123/newDGFix

Standardize developer guide capitalisation

ranaldmiao committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-W17-4/main into updateIndentation

sijie123 committed:
Fix links and capitalisation

sijie123 committed:
Merge pull request #343 from sijie123/newDGFix

Fix links and capitalisation

ranaldmiao committed:
Merge branch 'master' of github.com:CS2103-AY1819S1-W17-4/main into updateIndentation

sijie123 committed:
Update sijie PPP

ranaldmiao committed:
Fix Ranald's ppp for typo

sijie123 committed:
Merge pull request #345 from sijie123/newDGFix

Update sijie PPP

ranaldmiao committed:
Merge pull request #315 from sijie123/updateIndentation

Update indentation

ranaldmiao committed:
Merge pull request #344 from ranaldmiao/ppp-fix

Fix Ranald's ppp for typo

sijie123 committed:
Bump version number

sidhant007 committed:
Shit

ranaldmiao committed:
Merge pull request #346 from sijie123/lastonepls

Bump version number

sidhant007 committed:
Merge https://github.com/CS2103-AY1819S1-W17-4/main

sidhant007 committed:
Add kien stuff back

ranaldmiao committed:
Merge pull request #347 from sidhant007/kien-stuff2

Kien stuff2
